doc/rst/control_eps.rst

```diff
@@ -0,0 +1,145 @@
+
+|newpage|
+
+.. _sec_control_eps:
+
+*****************
+Control Endpoints
+*****************
+
+``lib_xud`` provides helper functions that provide a set of standard functionality to aid the
+creation of USB devices.
+
+Control transfers are typically used for command and status operations. They are essential to set
+up a USB device with all enumeration functions being performed using Control transfers. Control
+transfers are characterised by the use of a `SETUP` transaction.
+
+USB devices must provide an implementation of a control endpoint at endpoint 0.
+Endpoint 0 must deal with enumeration and configuration requests from the host. Many enumeration
+requests are compulsory and common to all devices, with most of them being requests for mandatory
+descriptors (Configuration, Device, String, etc).
+
+Since these requests are common across most (if not all) devices, some useful functions are
+provided to deal with them.
+
+Helper Functions
+================
+
+Firstly, the function ``USB_GetSetupPacket()`` is provided. This makes a call to the low-level
+`XUD` function ``XUD_GetSetupBuffer()`` with the 8 byte `Setup` packet which it parses into a
+``USB_SetupPacket_t`` structure for further inspection. The ``USB_SetupPacket_t`` structure passed
+by reference to ``USB_GetSetupPacket()`` is populated by the function.
+
+At this point the request is in a good state to be parsed by endpoint 0. Please see Universal
+Serial Bus 2.0 specification for full details of Setup packet and request structure.
+
+A ``USB_StandardRequests()`` function provides a bare-minimum implementation of the mandatory
+requests required to be implemented by a USB device.
+The function inspects this ``USB_SetupPacket_t`` structure and includes a minimum implementation
+of the Standard Device requests. The rest of this section documents the requests handled and a
+lists the basic  functionality associated with the request.
+
+It is not intended that this replace a good knowledge of the requests required, since the
```

Comment:
``USB_StandardRequests()`` has no knowledge about the specific class that is implemented, and hence does not guarantee a fully USB compliant device.